packages/teleterm/src/ui/DocumentGateway/useDocumentGateway.ts

```diff
@@ -75,6 +73,12 @@ export default function useGateway(doc: types.DocumentGateway) {
     }
   }, [disconnectAttempt.status]);
 
+  useEffect(() => {
+    if (cluster.connected) {
+      createGateway();
+    }
+  }, [cluster.connected]);
+
```

Comment:
What kind of scenario is this + the change in the `reconnect` above are going to handle?

Reply:
When you click on a connection it opens new tab with text: Click to reconnect, even when you are connected to the cluster (and in `useReconnect` is just changes document status).
After this change, clicking on a connection will try to make the _connection_ automatically and open a tab.